### PR TITLE
Cap Max Number of Pages Generated

### LINKED ISSUE #720: Place limit on how many pages can be added to a product
Currently, users can build what appears to be an infinite amount of pages, we should limit this since the application isn't meant for full-production type applications, but prototyping. Need to find a number to limit it to.

### PR DESCRIPTION
**Pull request by streetcoder123**
Tasks:
- https://microsoftgarage.visualstudio.com/Intern%20GitHub/_workitems/edit/30890
- https://microsoftgarage.visualstudio.com/Intern%20GitHub/_workitems/edit/30985

**Changes**
- Add 20 as the max number of pages allowed to be generated.
- Show a message to the user if he/she tries to add more than 20 pages.
- Abstract UI away from dependency checker component to be reused as a generic component (Notification).

closes #720

### CODE REVIEW

**Comment by jasmingoh** [comment]
Not sure if this is intended, but the warning sign looks like it is closer to the top than the bottom border
image

**Review by jasmingoh — DISMISSED** [review]

File: src/client/src/components/Notification/index.tsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  });
+
+  return (
+    <React.Fragment>
+      <div role="img" aria-label={altMessage}>
+        {!showWarning ? (
+          <Checkmark className={styles.iconCheck} />
+        ) : (
+          <Warning className={styles.iconWarning} />
+        )}
```
Comment (piotrmark) [inline src/client/src/components/Notification/index.tsx:31]: NIT: can we revert the condition to skip negation? Would be more readable IMO

File: src/client/src/containers/SelectOption/index.tsx [hunk 2]
```diff
@@ -13,12 +16,36 @@ import { ISelected } from "../../types/selected";
 import { Dispatch } from "redux";
 import RootAction from "../../actions/ActionType";
 
+import { ReactComponent as Warning } from "../../assets/warning.svg";
+import { ReactComponent as Checkmark } from "../../assets/checkgreen.svg";
```
Comment (piotrmark) [inline src/client/src/containers/SelectOption/index.tsx]: Do we need these imports here?

File: src/client/src/containers/SelectOption/index.tsx [hunk 3]
```diff
@@ -34,28 +61,30 @@ interface ISelectOptionProps {
...
   constructor(props: Props) {
     super(props);
     const { selectedCardIndices } = props;
     this.state = {
-      selectedCardIndices
+      selectedCardIndices,
+      maxPageReached: false,
+      description: props.intl.formatMessage(messages.limitedPages)
     };
+    this.addPage = this.addPage.bind(this);
```
Comment (piotrmark) [inline src/client/src/containers/SelectOption/index.tsx]: We could avoid this binding by using arrow properties:
https://www.freecodecamp.org/news/react-binding-patterns-5-approaches-for-handling-this-92c651b5af56/

File: src/client/src/containers/SelectOption/index.tsx [hunk 4]
```diff
@@ -263,11 +303,26 @@ class SelectOption extends React.Component<Props, ISelectOptionState> {
...
+      isPagesSelection,
+      intl
     } = this.props;
     return (
       <div>
         <Title>{title}</Title>
+        {isPagesSelection && (
+          <div
+            className={classnames(styles.description, {
+              [styles.borderGreen]: !this.state.maxPageReached,
```
Comment (piotrmark) [inline src/client/src/containers/SelectOption/index.tsx]: We could also destructure variables from state above

File: src/client/src/containers/SelectOption/index.tsx (hunk 2 above)
Comment (streetcoder123) [reply src/client/src/containers/SelectOption/index.tsx]: Nope, left by mistake. :p

File: src/client/src/containers/SelectOption/index.tsx (hunk 3 above)
Comment (streetcoder123) [reply src/client/src/containers/SelectOption/index.tsx]: That's good to know!

**Review by piotrmark — APPROVED** [review]
That was quick :D Thanks!

**Review by jasmingoh — APPROVED** [review]

**Review by annielovesjs — APPROVED** [review]

**Pull request merged**